### ISSUE TITLE
since() query returning empty response

### ISSUE THREAD

**Issue by alen125** [Webklex/laravel-imap#215]
Hello,  
  
I have an issue while querying messages by date from the server, and I get an empty result when there should be new messages.
Here is the example of it:  
  
```php  
$messages = $this->emailClient  
                 ->getFolder("Inbox")  
                 ->query()  
                 ->since('28.04.2019')  
                 ->get();  
```  
  
I've tried several ways of doing this. Some of them are:
- sending Carbon instance in `since()` method e.g. `->since(Carbon::now()->subDays(2))`
- building custom query  using `where()` method with `SINCE` key e.g. `->where(['SINCE', Carbon::parse('15.03.2018')])`

but none of them work. In the above example, if I remove `since()` method everything works and I get all messages from that folder.

The only way that I could do this now is to get all messages from that folder and then parse it, which takes a lot of time since there are many messages in that folder.
  
I'm connecting to IMAP Outlook server for this purpose.  
IMAP Package version is "1.4.1".
Running PHP 7.2 on Ubuntu.
  
Is there another way or am I doing something wrong?

Other than that, everything works fine.

**Comment by JonahKlimack** [comment]
same problem here as well:

```
		$date = date('d.m.Y', time()-(86400*1));
		$mailsIds = $mailbox->searchMailbox('UNSEEN SINCE "'.$date.'"');
		dump($mailsIds);
		foreach($mailsIds as $key => $mailId)
		{
			dump($key);
			// dump($mailId);

			$message = $mailbox->getMail($mailId, true);
			dump($message);
		}
```

if the date is in a different format, like 14 May 2019 then i'll get an error message:

```
  2103 => 1649513
  2104 => 1649514
  2105 => 1649515
  2106 => 1649516
  2107 => 1649517
  2108 => 1649518
]
0
Whoops, looks like something went wrong.
1/1
ErrorException in Mailbox.php line 539:
imap_fetchheader(): Bad message number

```

**Comment by todorowww** [comment]
Date format for SINCE should be DD-MMM-YYYY, IMAP protocol doesn't accept just any format. 

From IMAP specs:

>date            = date-text / DQUOTE date-text DQUOTE
>
>date-day        = 1*2DIGIT
>                    ; Day of month
>
>date-month      = "Jan" / "Feb" / "Mar" / "Apr" / "May" / "Jun" /
>                  "Jul" / "Aug" / "Sep" / "Oct" / "Nov" / "Dec"
>
>date-text       = date-day "-" date-month "-" date-year
>
>date-year       = 4DIGIT

This works for me:
`
$messages = $folder->query()->since($since->format('d-M-Y'))->get()
`

**Comment by Webklex** [comment]
Hi @todorowww ,
I've experienced that only "d M y" works for me. Guess there are multiple "correct" formats?

If you take a look at https://tools.ietf.org/html/rfc2822 section "3.3. Date and Time Specification" which is referred by https://tools.ietf.org/html/rfc1064 "d M y"  should be used but rfc1064 provides an example with "d-M-y". Now it seems like "d-M-Y" is valid as well. Where didi you find "d-M-Y"? :)

Best regards

**Comment by alen125** [comment]
@todorowww solution worked for me
> This works for me:
> `$messages = $folder->query(null)->since($since->format('d-M-Y'))->get();`

I've tested this by sending request on Outlook server

Also I've added time according to rfc1064 like this
```php
$messages = $folder->query(null)->since($since->format('d-M-Y H:i:s-z'))->get();
```

Thanks :)

**Comment by lomotech** [comment]
`query(null)` is work for me, but why?

**Comment by Webklex** [comment]
Hi @lomotech ,
that's a windows feature :)

If you want to know more about it, please head over to these issues: #153 #100 #78

Best regards

**Comment by todorowww** [comment]
Hey @Webklex 
Yes, Windows/Outlook implementations of IMAP are indeed fun, and full of "features". My latest project has to connect to Office365 servers, and it was quite the ride. For all the projects where we had the need for IMAP, we used d-M-Y and it always worked only with this format. I guess it really does come down to IMAP implementation, and there are multiple correct formats.

I've found the data in the [RFC3501 - IMAP v4r1](https://tools.ietf.org/html/rfc3501) ([page 84](https://tools.ietf.org/html/rfc3501#page-84) has the date spec)

What I think is the "issue", is that SINCE uses internal date, instead of RFC2822 date in the headers. And obviously, some implementations are perhaps stricter, or looser, in upholding the standard.
>SENTSINCE \<date\> Messages whose [RFC-2822] Date: header (disregarding time and timezone) is within or later than the specified date.
>
>SINCE \<date\> Messages whose internal date (disregarding time and timezone) is within or later than the specified date.

> 2.3.3.  Internal Date Message Attribute
>The internal date and time of the message on the server.  This is not the date and time in the [RFC-2822] header, but rather a date and time which reflects when the message was received.
